src/modules/firewall/src/lib/Firewall.h

```diff
@@ -39,32 +40,167 @@ class FirewallLog
     static OSCONFIG_LOG_HANDLE m_logHandle;
 };
 
+class GenericRule
+{
+public:
+    enum class State
+    {
+        Present,
+        Absent
+    };
+
+    enum class Action
+    {
+        Accept,
+        Drop,
+        Reject
+    };
+
+    enum class Direction
+    {
+        In,
+        Out
+    };
```

Comment:
Why not arrays of string constants that would be directly used for parsing and assembling payloads? These enums are no different from the plain enums used before, they still need conversion to string constants. See the other comment for an example.